### PR TITLE
Extend constraints to constrainable patterns

### PR DESCRIPTION
**Pull request by uysalere**
Fix for #66.

I didn't add an extension on MediaTrackCapabilities as ranges don't really make sense for these properties.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -259,13 +260,62 @@ <h2 id="display-constraints">
         </p>
         <p>
           The <code><a data-link-for="MediaTrackConstraintSet">cursor</a></code> constraint
-          allows an application to specify if the cursor should be included in the captured 
+          allows an application to observe if the cursor is be included in the captured
```
Comment (martinthomson) [inline index.html]: Why this change?  Control over the inclusion of a cursor is a property we explicitly wanted to allow.  It doesn't affect source selection, so it's safe.  This change would revert an agreed change.

**Review by martinthomson — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (uysalere) [reply index.html]: Current state of the spec does not allow constraints, which is contradicting this statement about selecting cursor. It would make sense after #73 lands though.

Comment (jan-ivar) [reply index.html]: https://github.com/w3c/mediacapture-screen-share/pull/73 has merged. I agree with @martinthomson that the cursor constraint is neither read-only nor a security concern, so we should revert this part.

File: index.html [hunk 2]
```diff
@@ -259,13 +260,62 @@ <h2 id="display-constraints">
...
+            </dd>
+          </dl>
+        </section>
         <section>
           <h2>
             Extensions to <a class="idlType">MediaTrackConstraintSet</a>
           </h2>
+          <p>
+            <a href="https://www.w3.org/TR/mediacapture-streams/#idl-def-MediaTrackConstraintSet">
+            MediaTrackConstraintSet</a> is used for reading the current status of constraints.
```
Comment (jan-ivar) [inline index.html:317]: It is used both for reading and applying.

File: index.html [hunk 3]
```diff
@@ -300,10 +350,60 @@ <h2>
...
+            <dt>
+               <dfn><code>cursor</code></dfn> of type <span class=
+               "idlMemberType"><a>DOMString</a></span>
+            </dt>
+            <dd>
+              <p>
+                Assumes values from the <code><a>CursorCaptureConstraint</a></code> enumeration that
+                determines if and when the cursor is included in the captured display surface.
+              </p>
+            </dd>
```
Comment (jan-ivar) [inline index.html:406]: I believe we also need extensions to [MediaTrackCapabilities](https://w3c.github.io/mediacapture-main/getusermedia.html#dom-mediatrackcapabilities) for this to be complete.

**Review by jan-ivar — CHANGES_REQUESTED** [review]
Great to see this added. A few missing bits, and reverting the cursor constraint bit.

File: index.html (hunk 3 above)
Comment (jan-ivar) [reply index.html:406]: > I didn't add an extension on MediaTrackCapabilities as ranges don't really make sense for these properties.

String-based and boolean constrainable properties have "ranges" in the form of arrays containing all valid values.

**Comment by henbos** [comment]
What is the use case of wanting to limit the user's choices?

File: index.html (hunk 1 above)
Comment (aboba) [reply index.html]: Reverted.

**Comment by jan-ivar** [comment]
@henbos Not sure I understand the question. Constraints in this spec do not limit the user's choice. This PR adds missing bits to correctly extend the constrainable pattern we're already using.

**Comment by jan-ivar** [comment]
I think we can merge this and add the MediaTrackCapabilities in a second PR.
